### ISSUE TITLE
Token Bucket

### ISSUE THREAD

**Issue by nmondal** [Ayan71/Blox-xyz#1]
There is an issue right here:
https://github.com/Ayan71/Blox-xyz/blob/main/Blox.xyz%20Task.txt#L52
Is that an issue? Is that not an issue?

Also read:
https://www.amazon.in/Java-Concurrency-Practice-1-Goetz/dp/9332576521

**Comment by Ayan71** [comment]
yes is that issue i want use Synchronized because! he potential issue of
thread safety and to prevent race conditions when accessing the shared
resource tokenBucket Here's why synchronization i use


public String call_me(String input) {
    long now = System.currentTimeMillis();
    long timeSinceLastCall = now - lastCallTime;

    synchronized (lock) {
        if (timeSinceLastCall < 60000) {
            if (tokenBucket.get() > 0) {
                tokenBucket.decrementAndGet();
            } else {
                try {
                    TimeUnit.MILLISECONDS.sleep(60000 - timeSinceLastCall);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
                tokenBucket.set(14);
            }
        } else {
            tokenBucket.set(14);
        }
    }

**Comment by nmondal** [comment]
@Ayan71 please think where it can go wrong. It is still wrong.

**Comment by Ayan71** [comment]
I have reviewed the code thoroughly and I think there are no issues or
problems with it !
